### PR TITLE
Feat/block failed batch

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

- Add disable state of the table (WATT)
- Disable the grid areas table when a batch has failed (DH)
- Ensure calculation steps won't be accessible when the batch has failed. (DH)
- Adjust wholesale mocks, so endpoints use the same dataset. (DH)

### CODE REVIEW

File: libs/ui-watt/src/lib/components/table/watt-table.component.html [hunk 1]
```diff
@@ -80,10 +80,10 @@
     mat-row
     *matRowDef="let row; columns: _getColumns()"
     [attr.aria-selected]="row === activeRow"
-    (click)="rowClick.emit(row)"
+    (click)="disabled ? null : rowClick.emit(row)"
     [ngClass]="{
-      'watt-table-highlight-row': !suppressRowHoverHighlight,
-      'watt-table-clickable-row': rowClick.observed,
+      'watt-table-highlight-row': !disabled || !suppressRowHoverHighlight,
```
Comment (ManBearTM) [inline libs/ui-watt/src/lib/components/table/watt-table.component.html:85]: question: shouldn't this be `&&`?

File: libs/ui-watt/src/lib/components/table/watt-table.component.html [hunk 2]
```diff
@@ -80,10 +80,10 @@
     mat-row
     *matRowDef="let row; columns: _getColumns()"
     [attr.aria-selected]="row === activeRow"
-    (click)="rowClick.emit(row)"
+    (click)="disabled ? null : rowClick.emit(row)"
```
Comment (ManBearTM) [inline libs/ui-watt/src/lib/components/table/watt-table.component.html:83]: suggestion: move this to class method, something like `_onRowClick`, seems a bit wrong to return null here IMO.

File: libs/ui-watt/src/lib/components/table/watt-table.component.ts [hunk 3]
```diff
@@ -159,6 +160,26 @@ export class WattTableComponent<T>
...
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  /**
+   * Used for disabling the table. This will disable all user interaction
+   */
+  set disabled(value: BooleanInput) {
+    this._disabled = coerceBooleanProperty(value);
+  }
```
Comment (ManBearTM) [inline libs/ui-watt/src/lib/components/table/watt-table.component.ts:181]: question: is it neccessary to coerce here as opposed to a simple boolean?

File: libs/ui-watt/src/lib/components/table/watt-table.component.html [hunk 4]
```diff
@@ -80,10 +80,10 @@
     mat-row
     *matRowDef="let row; columns: _getColumns()"
     [attr.aria-selected]="row === activeRow"
-    (click)="rowClick.emit(row)"
+    (click)="_onRowClick(row)"
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/table/watt-table.component.html:83]: nitpick: probably something we can't do much about right now, but even though the click doesn't have a visual effect, change detection will still be triggered every time there's a click on a row. That's most likely not an issue when the visual cues, like mouse pointer or row highlight, are not there, but still something to be aware of.

The best, of course, is if it was possible to not set the click in the first place, but I'm not sure there's an easy solution for that. :)

Comment (dzhavat) [reply libs/ui-watt/src/lib/components/table/watt-table.component.html:83]: What do you mean by adding it manually?

Comment (ManBearTM) [reply libs/ui-watt/src/lib/components/table/watt-table.component.html:83]: I feel like this is an optimization we can look into **if** it becomes relevant. But it is a good observation!

**Review by ManBearTM — APPROVED** [review]

File: libs/ui-watt/src/lib/components/table/watt-table.component.html (hunk 4 above)
Comment (dzhavat) [reply libs/ui-watt/src/lib/components/table/watt-table.component.html:83]: Yes, for sure. Nothing

**Pull request merged**